src/common/state/redis.c

```diff
@@ -123,6 +123,60 @@ void db_attach(db_handle *db, event_loop *loop) {
   redisAeAttach(loop, db->sub_context);
 }
 
+/**
+ * An internal function to allocate a task object and parse a hashmap reply
+ * from Redis into the task object.  If the Redis reply is malformed, an empty
+ * task with the given task ID is returned.
+ *
+ * @param id The ID of the task we're looking up. If the reply from Redis is
+ *        well-formed, the reply's ID should match this ID. Else, the returned
+ *        task will have its ID set to this ID.
+ * @param num_redis_replies The number of keys and values in the Redis hashmap.
+ * @param redis_replies A pointer to the Redis hashmap keys and values.
+ * @return A pointer to the parsed task.
+ */
+task *parse_redis_task_table_entry(task_id id,
+                                   int num_redis_replies,
+                                   redisReply **redis_replies) {
+  task *task_result;
+  if (num_redis_replies == 0) {
+    /* If there was no information about this task, an empty task. */
+    task_result = alloc_nil_task(id);
```

Comment:
No, unless we decide to do the update commands to the task and result tables in parallel. I'd like to keep it non-fatal so we can test that the error triggers, but I'll change it to return NULL.